ccm_web/client/src/components/MultipleUserEnrollmentWorkflow.tsx

```diff
@@ -88,14 +100,50 @@ export default function MultipleUserEnrollmentWorkflow (props: MultipleUserEnrol
   const [schemaInvalidations, setSchemaInvalidations] = useState<SchemaInvalidation[] | undefined>(undefined)
   const [rowInvalidations, setRowInvalidations] = useState<EnrollmentInvalidation[] | undefined>(undefined)
 
+  const [processResult, setProcessResult] = useState<APIProcessResult | undefined>(undefined)
+
   const [
-    doAddExternalEnrollments, isAddExternalEnrollmentsLoading, addExternalEnrollmentsError, clearAddExternalEnrollmentsError
+    doAddExternalEnrollments, isAddExternalEnrollmentsLoading, addExternalEnrollmentsError,
+    clearAddExternalEnrollmentsError
   ] = usePromise(
-    async (sectionId: number, enrollments: AddNewExternalUserEnrollment[]) => {
-      const promise = new Promise(resolve => setTimeout(resolve, 3000)) // Mocking this for now
-      return await promise
+    async (sectionId: number, enrollments: AddNewExternalUserEnrollment[]): Promise<APIProcessResult> => {
+      let successes: ExternalUserSuccess[]
+      const errors: ErrorDescription[] = []
+      try {
+        successes = await api.createExternalUsers(
+          enrollments.map(e => ({ email: e.email, givenName: e.firstName, surname: e.lastName }))
+        )
+      } catch (error: unknown) {
+        if (error instanceof ExternalUserProcessError) {
+          errors.push(...error.describeErrors())
+          successes = error.data.filter(r => isExternalUserSuccess(r)) as ExternalUserSuccess[]
```

Comment:
When creating a single user and Cirrus failure, I see this response 
`{"statusCode":400,"data":[{"email":"dfdsf@dsdsd.com","userCreated":true,"invited":{"statusCode":400,"messages":["Could not save file content to S3"]}}]}`

End user output

![Screen Shot 2022-03-15 at 2 55 10 PM](https://user-images.githubusercontent.com/8579775/158451082-b50c5889-265a-4341-8913-13b97bd84a41.png)

Reply:
In case of multiple user cirrus failure and creation of user a success, below is the JSON response

Note: Cirrus seems to be down for some reason today. 

```
{"statusCode":400,"data":[{"email":"1000@emai.edu","userCreated":true,"invited":{"statusCode":400,"messages":["Could not save file content to S3"]}},{"email":"1002@emai.edu","userCreated":true,"invited":{"statusCode":400,"messages":["Could not save file content to S3"]}},{"email":"1003@gmai.edu","userCreated":true,"invited":{"statusCode":400,"messages":["Could not save file content to S3"]}},{"email":"1004@gmai.edu","userCreated":true,"invited":{"statusCode":400,"messages":["Could not save file content to S3"]}},{"email":"1005@gmai.edu","userCreated":true,"invited":{"statusCode":400,"messages":["Could not save file content to S3"]}},{"email":"1006@gmai.edu","userCreated":true,"invited":{"statusCode":400,"messages":["Could not save file content to S3"]}},{"email":"1007@gmai.edu","userCreated":true,"invited":{"statusCode":400,"messages":["Could not save file content to S3"]}},{"email":"1008@gma.edu","userCreated":true,"invited":{"statusCode":400,"messages":["Could not save file content to S3"]}}]}
```
![Screen Shot 2022-03-15 at 3 04 15 PM](https://user-images.githubusercontent.com/8579775/158452459-143d2276-a8b1-4c6d-93a6-7de23d37de0c.png)

Reply:
I have reported this issue with Cirrus yesterday, so they are looking into it.

Reply:
OK, Cirrus has a fix on their end for this error. Back to business!

Reply:
OK, Cirrus has a fix on their end for this error. Back to business!